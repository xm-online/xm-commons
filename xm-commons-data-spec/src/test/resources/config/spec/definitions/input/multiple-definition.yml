---
test-items:
  - key: store/GET-EMPLOYEES-AGE
    inputSpec: |
      {
        "$schema": "http://json-schema.org/draft-07/schema#",
        "type": "object",
        "properties": {
          "employeesInfo": {
            "type": "array",
            "items": {
              "type": "object",
              "properties": {
                "employeeInfo": {
                  "$ref": "#/xmDefinition/employeeInfo"
                }
              }
            }
          },
          "location": {
            "$ref": "#/xmDefinition/address"
          }
        },
        "required": ["employeesInfo"]
      }
definitions:
  - key: address
    ref: json-definitions/address.json
  - key: geoAddress
    ref: json-definitions/geoAddress.json
  - key: employeeInfo
    value: |-
      {
        "$schema": "http://json-schema.org/draft-07/schema#",
        "type": "object",
        "properties": {
          "storeName": {
            "type": "string"
          },
          "info": {
            "$ref": "#/xmDefinition/userInfo"
          }
        }
      }
  - key: userInfo
    value: |-
      {
        "$schema": "http://json-schema.org/draft-07/schema#",
        "type": "object",
        "properties": {
          "username": {
            "type": "string"
          },
          "phoneNumber": {
            "type": "string"
          },
          "age": {
            "type": "number"
          }
        }
      }
