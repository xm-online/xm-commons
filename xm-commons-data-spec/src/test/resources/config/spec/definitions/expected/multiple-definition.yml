---
test-items:
  - key: store/GET-EMPLOYEES-AGE
    inputSpec: |
        {
          "$schema":"http://json-schema.org/draft-07/schema#",
          "type":"object",
          "properties":{
              "employeesInfo":{
                  "type":"array",
                  "items":{
                      "type":"object",
                      "properties":{
                          "employeeInfo":{
                              "$ref":"#/xmDefinition/employeeInfo"
                          }
                      }
                  }
              },
              "location":{
                  "$ref":"#/xmDefinition/address"
              }
          },
          "required":[
              "employeesInfo"
          ],
          "xmDefinition":{
              "address":{
                  "type":"object",
                  "properties":{
                      "address":{
                          "type":"string"
                      },
                      "geoAddress":{
                          "$ref":"#/xmDefinition/geoAddress"
                      },
                      "formattedAddress":{
                          "type":"string"
                      }
                  }
              },
              "geoAddress":{
                  "type":"object",
                  "properties":{
                      "location":{
                          "lat":{
                              "type":"number"
                          },
                          "lng":{
                              "type":"number"
                          }
                      },
                      "place_id":{
                          "type":"string"
                      },
                      "utcOffset":{
                          "type":"integer"
                      },
                      "components":{
                          "type":"object"
                      }
                  }
              },
              "employeeInfo":{
                  "$schema":"http://json-schema.org/draft-07/schema#",
                  "type":"object",
                  "properties":{
                      "storeName":{
                          "type":"string"
                      },
                      "info":{
                          "$ref":"#/xmDefinition/userInfo"
                      }
                  }
              },
              "userInfo":{
                  "$schema":"http://json-schema.org/draft-07/schema#",
                  "type":"object",
                  "properties":{
                      "username":{
                          "type":"string"
                      },
                      "phoneNumber":{
                          "type":"string"
                      },
                      "age":{
                          "type":"number"
                      }
                  }
              }
          }
        }
definitions:
  - key: address
    ref: json-definitions/address.json
  - key: geoAddress
    ref: json-definitions/geoAddress.json
  - key: employeeInfo
    value: |-
      {
        "$schema": "http://json-schema.org/draft-07/schema#",
        "type": "object",
        "properties": {
          "storeName": {
            "type": "string"
          },
          "info": {
            "$ref": "#/xmDefinition/userInfo"
          }
        }
      }
  - key: userInfo
    value: |-
      {
        "$schema": "http://json-schema.org/draft-07/schema#",
        "type": "object",
        "properties": {
          "username": {
            "type": "string"
          },
          "phoneNumber": {
            "type": "string"
          },
          "age": {
            "type": "number"
          }
        }
      }
